libraries/abstractions/platform/freertos/include/platform/iot_network_udp.h

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (C) 2019 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file iot_network_udp.h
+ * @brief Declares the network stack functions specified in platform/iot_network.h for
+ * Amazon FreeRTOS UDP Sockets.
+ */
+
+#ifndef _IOT_NETWORK_UDP_H_
+#define _IOT_NETWORK_UDP_H_
+
+/* Standard includes. */
+#include <stdbool.h>
+
+/* Platform network include. */
+#include "platform/iot_network.h"
+
+/* Amazon FreeRTOS Secure Sockets include. */
+#include "iot_secure_sockets.h"
+
+/**
+ * @brief Represents a network connection that uses Amazon FreeRTOS UDP Sockets.
+ *
+ * This is an incomplete type. In application code, only pointers to this type
+ * should be used.
+ */
+typedef struct _networkConnection IotNetworkConnectionUdp_t;
+
+/**
+ * @brief Information on the remote server for connection setup.
+ *
+ * Passed to #IotNetworkUdp_Create as `pConnectionInfo`.
+ *
+ * All instances of #IotNetworkServerInfoUdp_t should be initialized with
+ * #IOT_NETWORK_SERVER_INFO_UDP_INITIALIZER.
+ * @Note If noBind is set to true, underlying stack must support bind in send
+ */
+typedef struct IotNetworkServerInfoUdp
+{
+    const char * pHostName;     /**< @brief Server host name. Must be NULL-terminated. */
```

Comment:
Per above, our UDP network interface should be based on Berkeley, including the standard socket options.